mlflow/utils/rest_utils.py

```diff
@@ -60,9 +62,14 @@ def http_request(
 
     :return: requests.Response object.
     """
-    max_retries = max_retries or MLFLOW_HTTP_REQUEST_MAX_RETRIES.get()
-    backoff_factor = backoff_factor or MLFLOW_HTTP_REQUEST_BACKOFF_FACTOR.get()
-    timeout = timeout or MLFLOW_HTTP_REQUEST_TIMEOUT.get()
+    max_retries = MLFLOW_HTTP_REQUEST_MAX_RETRIES.get() if max_retries is None else max_retries
```

Comment:
In the original code, `max_retries = 0` falls back to `MLFLOW_HTTP_REQUEST_MAX_RETRIES.get()` but it should not.